{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2024-03-02 18:34:12--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
      "Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4\n",
      "Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 10249221 (9.8M) [application/zip]\n",
      "Saving to: ‘fastqc_v0.11.9.zip’\n",
      "\n",
      "fastqc_v0.11.9.zip  100%[===================>]   9.77M  13.0MB/s    in 0.8s    \n",
      "\n",
      "2024-03-02 18:34:13 (13.0 MB/s) - ‘fastqc_v0.11.9.zip’ saved [10249221/10249221]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "unzip:  cannot find or open fastqc_v0.11.9.zip, fastqc_v0.11.9.zip.zip or fastqc_v0.11.9.zip.ZIP.\n",
      "rm: cannot remove 'fastqc_v0.11.9.zip': No such file or directory\n"
     ]
    }
   ],
   "source": [
    "!unzip fastqc_v0.11.9.zip\n",
    "!rm fastqc_v0.11.9.zip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "!chmod a+x FastQC/fastqc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Обработка последовательностей"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2024-03-02 18:10:21--  https://www.encodeproject.org/files/ENCFF000AXS/@@download/ENCFF000AXS.fastq.gz\n",
      "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
      "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
      "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
      "Location: https://encode-public.s3.amazonaws.com/2010/11/16/e7586585-85b6-4b9c-a85b-07433a82dee5/ENCFF000AXS.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF000AXS.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXTEDJVEHW&Signature=nLZRYXXSZKH10wxdVmS7QDaOXDQ%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEE8aCXVzLXdlc3QtMiJHMEUCIQCg8I3adeArV2%2B0PKr%2BliReQtlFeMj6i3AHzT3qXna6eAIgbZeL3IvGujRMpEboaw8MwQow0EGhyw2tOJXKgg3doacqswUISBAAGgwyMjA3NDg3MTQ4NjMiDCgUfDle81%2BwRGfz%2BCqQBV6cMlUvhVIWEsktBTfDwfLQzodwFBRxWo%2FItlsYXhdAiHewbnTFxE61yy8HwQqlN5Ow1de3bJuvwLnpcVAtXIG6VQUcNqZ%2FiU%2FhEc0q%2FEzM%2B%2BaXXxnf%2FUUeEEmzTN4nzeg4KndDo%2FMWI%2FWZ8BTR%2FxLQAWlsnNPZJyOI6JgEjz%2BU%2BN1qSh7gtmN8uPQtxbbeN5bTjwN57syOOqvKXf17mLOEKIVOARNHzgh5%2FqNw4rR1Wt0jSDKG0CofViDs5b%2FG35Z1%2BUzIqFDQakEJT%2BEgFbFQ930hFpGPgQq6Ww8sP9lYwdO4d06V%2Bf74vJ6kMCUGXf0g37P5RAg9MfmT49x9CthNB16MxwCCItFpJMrkK9v2cwUfZSgXku09sMzoUaVEI3XSLZMeZ%2FJQzUx6NkhtsLxjlJwQFOAPawN6MXiRNJHqLjzd8eC7fwidO1cA2e%2BsN%2BRl6oZezQ%2BGoSn9sipubEy19BS8Q3f26taMudvgvm%2BD4iGNi2O%2FlusOIgF%2B9blPB%2FqBiJqapu%2F2fvygxzjh2IBHwhvPndqwSP86SdaxMTv0wrUZMpotGmSMvdIGCmFQNdg534wlLECtd6Xqt1b4sagFr2jG7nefZvThD%2FuHSxRFVcaHcNRG2UdBEFzE4RmF1V58cXmu%2F59tyJFiYYwB8IIacTg6CaPpYPz1u24R7z77WK3Eu5d2uPR2hk7U4lxFhxg3YfMZvyxXXhNj%2BBMsACuHnXDkSPiZYi%2BhdGP1dCGHzXSORef0ioYPS7pUkZoEiy4YRQZn4rkTrXphi%2Fwi0uvsKY4k%2BxFuc9drFHhFT%2BiGOfV6LfNUluhUjP%2BJULIydLxibgV7hvcKGC8WMylFskTudVBA3QMISaVbYOpKsRyqMOHujK8GOrEB%2BBYceWJkR6og5qOcSsCeOTgAsi4goEzwS%2Bdx0TuSNHauuoccZM9JMVledPGMyxjCTipjf36cZ0%2BxMujS%2B2q4Qr1iMDDtAAKfwY7W5DC5ltgViWFfHjg00641r1i2gyYDBk2QzQjyyFNS60JXGsHvN0MwylvU3g8GXJ7OqNfGCEYlZs3cSyo5Ddic4TiwpSY%2Ba%2B1Aemag%2B5GzxiIQVzkXeXVYbuWpzItoxWxp6b%2Fs1NXh&Expires=1709521824 [following]\n",
      "--2024-03-02 18:10:22--  https://encode-public.s3.amazonaws.com/2010/11/16/e7586585-85b6-4b9c-a85b-07433a82dee5/ENCFF000AXS.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF000AXS.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXTEDJVEHW&Signature=nLZRYXXSZKH10wxdVmS7QDaOXDQ%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEE8aCXVzLXdlc3QtMiJHMEUCIQCg8I3adeArV2%2B0PKr%2BliReQtlFeMj6i3AHzT3qXna6eAIgbZeL3IvGujRMpEboaw8MwQow0EGhyw2tOJXKgg3doacqswUISBAAGgwyMjA3NDg3MTQ4NjMiDCgUfDle81%2BwRGfz%2BCqQBV6cMlUvhVIWEsktBTfDwfLQzodwFBRxWo%2FItlsYXhdAiHewbnTFxE61yy8HwQqlN5Ow1de3bJuvwLnpcVAtXIG6VQUcNqZ%2FiU%2FhEc0q%2FEzM%2B%2BaXXxnf%2FUUeEEmzTN4nzeg4KndDo%2FMWI%2FWZ8BTR%2FxLQAWlsnNPZJyOI6JgEjz%2BU%2BN1qSh7gtmN8uPQtxbbeN5bTjwN57syOOqvKXf17mLOEKIVOARNHzgh5%2FqNw4rR1Wt0jSDKG0CofViDs5b%2FG35Z1%2BUzIqFDQakEJT%2BEgFbFQ930hFpGPgQq6Ww8sP9lYwdO4d06V%2Bf74vJ6kMCUGXf0g37P5RAg9MfmT49x9CthNB16MxwCCItFpJMrkK9v2cwUfZSgXku09sMzoUaVEI3XSLZMeZ%2FJQzUx6NkhtsLxjlJwQFOAPawN6MXiRNJHqLjzd8eC7fwidO1cA2e%2BsN%2BRl6oZezQ%2BGoSn9sipubEy19BS8Q3f26taMudvgvm%2BD4iGNi2O%2FlusOIgF%2B9blPB%2FqBiJqapu%2F2fvygxzjh2IBHwhvPndqwSP86SdaxMTv0wrUZMpotGmSMvdIGCmFQNdg534wlLECtd6Xqt1b4sagFr2jG7nefZvThD%2FuHSxRFVcaHcNRG2UdBEFzE4RmF1V58cXmu%2F59tyJFiYYwB8IIacTg6CaPpYPz1u24R7z77WK3Eu5d2uPR2hk7U4lxFhxg3YfMZvyxXXhNj%2BBMsACuHnXDkSPiZYi%2BhdGP1dCGHzXSORef0ioYPS7pUkZoEiy4YRQZn4rkTrXphi%2Fwi0uvsKY4k%2BxFuc9drFHhFT%2BiGOfV6LfNUluhUjP%2BJULIydLxibgV7hvcKGC8WMylFskTudVBA3QMISaVbYOpKsRyqMOHujK8GOrEB%2BBYceWJkR6og5qOcSsCeOTgAsi4goEzwS%2Bdx0TuSNHauuoccZM9JMVledPGMyxjCTipjf36cZ0%2BxMujS%2B2q4Qr1iMDDtAAKfwY7W5DC5ltgViWFfHjg00641r1i2gyYDBk2QzQjyyFNS60JXGsHvN0MwylvU3g8GXJ7OqNfGCEYlZs3cSyo5Ddic4TiwpSY%2Ba%2B1Aemag%2B5GzxiIQVzkXeXVYbuWpzItoxWxp6b%2Fs1NXh&Expires=1709521824\n",
      "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.179.17, 52.92.235.145, 52.92.160.41, ...\n",
      "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.179.17|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 251147129 (240M) [binary/octet-stream]\n",
      "Saving to: ‘ENCFF000AXS.fastq.gz’\n",
      "\n",
      "ENCFF000AXS.fastq.g 100%[===================>] 239.51M  3.34MB/s    in 76s     \n",
      "\n",
      "2024-03-02 18:11:39 (3.14 MB/s) - ‘ENCFF000AXS.fastq.gz’ saved [251147129/251147129]\n",
      "\n",
      "--2024-03-02 18:11:40--  https://www.encodeproject.org/files/ENCFF000AXZ/@@download/ENCFF000AXZ.fastq.gz\n",
      "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
      "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
      "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
      "Location: https://encode-public.s3.amazonaws.com/2010/11/16/fb28f2dd-4ae7-47c9-9dce-309e710fe451/ENCFF000AXZ.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF000AXZ.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXTEDJVEHW&Signature=6Bah1sHtgE%2BxtXi6n5MsGC77OCA%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEE8aCXVzLXdlc3QtMiJHMEUCIQCg8I3adeArV2%2B0PKr%2BliReQtlFeMj6i3AHzT3qXna6eAIgbZeL3IvGujRMpEboaw8MwQow0EGhyw2tOJXKgg3doacqswUISBAAGgwyMjA3NDg3MTQ4NjMiDCgUfDle81%2BwRGfz%2BCqQBV6cMlUvhVIWEsktBTfDwfLQzodwFBRxWo%2FItlsYXhdAiHewbnTFxE61yy8HwQqlN5Ow1de3bJuvwLnpcVAtXIG6VQUcNqZ%2FiU%2FhEc0q%2FEzM%2B%2BaXXxnf%2FUUeEEmzTN4nzeg4KndDo%2FMWI%2FWZ8BTR%2FxLQAWlsnNPZJyOI6JgEjz%2BU%2BN1qSh7gtmN8uPQtxbbeN5bTjwN57syOOqvKXf17mLOEKIVOARNHzgh5%2FqNw4rR1Wt0jSDKG0CofViDs5b%2FG35Z1%2BUzIqFDQakEJT%2BEgFbFQ930hFpGPgQq6Ww8sP9lYwdO4d06V%2Bf74vJ6kMCUGXf0g37P5RAg9MfmT49x9CthNB16MxwCCItFpJMrkK9v2cwUfZSgXku09sMzoUaVEI3XSLZMeZ%2FJQzUx6NkhtsLxjlJwQFOAPawN6MXiRNJHqLjzd8eC7fwidO1cA2e%2BsN%2BRl6oZezQ%2BGoSn9sipubEy19BS8Q3f26taMudvgvm%2BD4iGNi2O%2FlusOIgF%2B9blPB%2FqBiJqapu%2F2fvygxzjh2IBHwhvPndqwSP86SdaxMTv0wrUZMpotGmSMvdIGCmFQNdg534wlLECtd6Xqt1b4sagFr2jG7nefZvThD%2FuHSxRFVcaHcNRG2UdBEFzE4RmF1V58cXmu%2F59tyJFiYYwB8IIacTg6CaPpYPz1u24R7z77WK3Eu5d2uPR2hk7U4lxFhxg3YfMZvyxXXhNj%2BBMsACuHnXDkSPiZYi%2BhdGP1dCGHzXSORef0ioYPS7pUkZoEiy4YRQZn4rkTrXphi%2Fwi0uvsKY4k%2BxFuc9drFHhFT%2BiGOfV6LfNUluhUjP%2BJULIydLxibgV7hvcKGC8WMylFskTudVBA3QMISaVbYOpKsRyqMOHujK8GOrEB%2BBYceWJkR6og5qOcSsCeOTgAsi4goEzwS%2Bdx0TuSNHauuoccZM9JMVledPGMyxjCTipjf36cZ0%2BxMujS%2B2q4Qr1iMDDtAAKfwY7W5DC5ltgViWFfHjg00641r1i2gyYDBk2QzQjyyFNS60JXGsHvN0MwylvU3g8GXJ7OqNfGCEYlZs3cSyo5Ddic4TiwpSY%2Ba%2B1Aemag%2B5GzxiIQVzkXeXVYbuWpzItoxWxp6b%2Fs1NXh&Expires=1709521902 [following]\n",
      "--2024-03-02 18:11:41--  https://encode-public.s3.amazonaws.com/2010/11/16/fb28f2dd-4ae7-47c9-9dce-309e710fe451/ENCFF000AXZ.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF000AXZ.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXTEDJVEHW&Signature=6Bah1sHtgE%2BxtXi6n5MsGC77OCA%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEE8aCXVzLXdlc3QtMiJHMEUCIQCg8I3adeArV2%2B0PKr%2BliReQtlFeMj6i3AHzT3qXna6eAIgbZeL3IvGujRMpEboaw8MwQow0EGhyw2tOJXKgg3doacqswUISBAAGgwyMjA3NDg3MTQ4NjMiDCgUfDle81%2BwRGfz%2BCqQBV6cMlUvhVIWEsktBTfDwfLQzodwFBRxWo%2FItlsYXhdAiHewbnTFxE61yy8HwQqlN5Ow1de3bJuvwLnpcVAtXIG6VQUcNqZ%2FiU%2FhEc0q%2FEzM%2B%2BaXXxnf%2FUUeEEmzTN4nzeg4KndDo%2FMWI%2FWZ8BTR%2FxLQAWlsnNPZJyOI6JgEjz%2BU%2BN1qSh7gtmN8uPQtxbbeN5bTjwN57syOOqvKXf17mLOEKIVOARNHzgh5%2FqNw4rR1Wt0jSDKG0CofViDs5b%2FG35Z1%2BUzIqFDQakEJT%2BEgFbFQ930hFpGPgQq6Ww8sP9lYwdO4d06V%2Bf74vJ6kMCUGXf0g37P5RAg9MfmT49x9CthNB16MxwCCItFpJMrkK9v2cwUfZSgXku09sMzoUaVEI3XSLZMeZ%2FJQzUx6NkhtsLxjlJwQFOAPawN6MXiRNJHqLjzd8eC7fwidO1cA2e%2BsN%2BRl6oZezQ%2BGoSn9sipubEy19BS8Q3f26taMudvgvm%2BD4iGNi2O%2FlusOIgF%2B9blPB%2FqBiJqapu%2F2fvygxzjh2IBHwhvPndqwSP86SdaxMTv0wrUZMpotGmSMvdIGCmFQNdg534wlLECtd6Xqt1b4sagFr2jG7nefZvThD%2FuHSxRFVcaHcNRG2UdBEFzE4RmF1V58cXmu%2F59tyJFiYYwB8IIacTg6CaPpYPz1u24R7z77WK3Eu5d2uPR2hk7U4lxFhxg3YfMZvyxXXhNj%2BBMsACuHnXDkSPiZYi%2BhdGP1dCGHzXSORef0ioYPS7pUkZoEiy4YRQZn4rkTrXphi%2Fwi0uvsKY4k%2BxFuc9drFHhFT%2BiGOfV6LfNUluhUjP%2BJULIydLxibgV7hvcKGC8WMylFskTudVBA3QMISaVbYOpKsRyqMOHujK8GOrEB%2BBYceWJkR6og5qOcSsCeOTgAsi4goEzwS%2Bdx0TuSNHauuoccZM9JMVledPGMyxjCTipjf36cZ0%2BxMujS%2B2q4Qr1iMDDtAAKfwY7W5DC5ltgViWFfHjg00641r1i2gyYDBk2QzQjyyFNS60JXGsHvN0MwylvU3g8GXJ7OqNfGCEYlZs3cSyo5Ddic4TiwpSY%2Ba%2B1Aemag%2B5GzxiIQVzkXeXVYbuWpzItoxWxp6b%2Fs1NXh&Expires=1709521902\n",
      "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.179.17, 52.92.235.145, 52.92.160.41, ...\n",
      "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.179.17|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 583880662 (557M) [binary/octet-stream]\n",
      "Saving to: ‘ENCFF000AXZ.fastq.gz’\n",
      "\n",
      "ENCFF000AXZ.fastq.g 100%[===================>] 556.83M  5.82MB/s    in 2m 54s  \n",
      "\n",
      "2024-03-02 18:14:36 (3.19 MB/s) - ‘ENCFF000AXZ.fastq.gz’ saved [583880662/583880662]\n",
      "\n",
      "--2024-03-02 18:14:36--  https://www.encodeproject.org/files/ENCFF659UFR/@@download/ENCFF659UFR.fastq.gz\n",
      "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
      "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
      "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
      "Location: https://encode-public.s3.amazonaws.com/2016/10/10/241ba9ba-2f87-42c2-a699-738277b03b4a/ENCFF659UFR.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF659UFR.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXTEDJVEHW&Signature=WivxhNN8b2Pgk%2BLzp6%2FZ3iGqB6I%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEE8aCXVzLXdlc3QtMiJHMEUCIQCg8I3adeArV2%2B0PKr%2BliReQtlFeMj6i3AHzT3qXna6eAIgbZeL3IvGujRMpEboaw8MwQow0EGhyw2tOJXKgg3doacqswUISBAAGgwyMjA3NDg3MTQ4NjMiDCgUfDle81%2BwRGfz%2BCqQBV6cMlUvhVIWEsktBTfDwfLQzodwFBRxWo%2FItlsYXhdAiHewbnTFxE61yy8HwQqlN5Ow1de3bJuvwLnpcVAtXIG6VQUcNqZ%2FiU%2FhEc0q%2FEzM%2B%2BaXXxnf%2FUUeEEmzTN4nzeg4KndDo%2FMWI%2FWZ8BTR%2FxLQAWlsnNPZJyOI6JgEjz%2BU%2BN1qSh7gtmN8uPQtxbbeN5bTjwN57syOOqvKXf17mLOEKIVOARNHzgh5%2FqNw4rR1Wt0jSDKG0CofViDs5b%2FG35Z1%2BUzIqFDQakEJT%2BEgFbFQ930hFpGPgQq6Ww8sP9lYwdO4d06V%2Bf74vJ6kMCUGXf0g37P5RAg9MfmT49x9CthNB16MxwCCItFpJMrkK9v2cwUfZSgXku09sMzoUaVEI3XSLZMeZ%2FJQzUx6NkhtsLxjlJwQFOAPawN6MXiRNJHqLjzd8eC7fwidO1cA2e%2BsN%2BRl6oZezQ%2BGoSn9sipubEy19BS8Q3f26taMudvgvm%2BD4iGNi2O%2FlusOIgF%2B9blPB%2FqBiJqapu%2F2fvygxzjh2IBHwhvPndqwSP86SdaxMTv0wrUZMpotGmSMvdIGCmFQNdg534wlLECtd6Xqt1b4sagFr2jG7nefZvThD%2FuHSxRFVcaHcNRG2UdBEFzE4RmF1V58cXmu%2F59tyJFiYYwB8IIacTg6CaPpYPz1u24R7z77WK3Eu5d2uPR2hk7U4lxFhxg3YfMZvyxXXhNj%2BBMsACuHnXDkSPiZYi%2BhdGP1dCGHzXSORef0ioYPS7pUkZoEiy4YRQZn4rkTrXphi%2Fwi0uvsKY4k%2BxFuc9drFHhFT%2BiGOfV6LfNUluhUjP%2BJULIydLxibgV7hvcKGC8WMylFskTudVBA3QMISaVbYOpKsRyqMOHujK8GOrEB%2BBYceWJkR6og5qOcSsCeOTgAsi4goEzwS%2Bdx0TuSNHauuoccZM9JMVledPGMyxjCTipjf36cZ0%2BxMujS%2B2q4Qr1iMDDtAAKfwY7W5DC5ltgViWFfHjg00641r1i2gyYDBk2QzQjyyFNS60JXGsHvN0MwylvU3g8GXJ7OqNfGCEYlZs3cSyo5Ddic4TiwpSY%2Ba%2B1Aemag%2B5GzxiIQVzkXeXVYbuWpzItoxWxp6b%2Fs1NXh&Expires=1709522079 [following]\n",
      "--2024-03-02 18:14:37--  https://encode-public.s3.amazonaws.com/2016/10/10/241ba9ba-2f87-42c2-a699-738277b03b4a/ENCFF659UFR.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF659UFR.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXTEDJVEHW&Signature=WivxhNN8b2Pgk%2BLzp6%2FZ3iGqB6I%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEE8aCXVzLXdlc3QtMiJHMEUCIQCg8I3adeArV2%2B0PKr%2BliReQtlFeMj6i3AHzT3qXna6eAIgbZeL3IvGujRMpEboaw8MwQow0EGhyw2tOJXKgg3doacqswUISBAAGgwyMjA3NDg3MTQ4NjMiDCgUfDle81%2BwRGfz%2BCqQBV6cMlUvhVIWEsktBTfDwfLQzodwFBRxWo%2FItlsYXhdAiHewbnTFxE61yy8HwQqlN5Ow1de3bJuvwLnpcVAtXIG6VQUcNqZ%2FiU%2FhEc0q%2FEzM%2B%2BaXXxnf%2FUUeEEmzTN4nzeg4KndDo%2FMWI%2FWZ8BTR%2FxLQAWlsnNPZJyOI6JgEjz%2BU%2BN1qSh7gtmN8uPQtxbbeN5bTjwN57syOOqvKXf17mLOEKIVOARNHzgh5%2FqNw4rR1Wt0jSDKG0CofViDs5b%2FG35Z1%2BUzIqFDQakEJT%2BEgFbFQ930hFpGPgQq6Ww8sP9lYwdO4d06V%2Bf74vJ6kMCUGXf0g37P5RAg9MfmT49x9CthNB16MxwCCItFpJMrkK9v2cwUfZSgXku09sMzoUaVEI3XSLZMeZ%2FJQzUx6NkhtsLxjlJwQFOAPawN6MXiRNJHqLjzd8eC7fwidO1cA2e%2BsN%2BRl6oZezQ%2BGoSn9sipubEy19BS8Q3f26taMudvgvm%2BD4iGNi2O%2FlusOIgF%2B9blPB%2FqBiJqapu%2F2fvygxzjh2IBHwhvPndqwSP86SdaxMTv0wrUZMpotGmSMvdIGCmFQNdg534wlLECtd6Xqt1b4sagFr2jG7nefZvThD%2FuHSxRFVcaHcNRG2UdBEFzE4RmF1V58cXmu%2F59tyJFiYYwB8IIacTg6CaPpYPz1u24R7z77WK3Eu5d2uPR2hk7U4lxFhxg3YfMZvyxXXhNj%2BBMsACuHnXDkSPiZYi%2BhdGP1dCGHzXSORef0ioYPS7pUkZoEiy4YRQZn4rkTrXphi%2Fwi0uvsKY4k%2BxFuc9drFHhFT%2BiGOfV6LfNUluhUjP%2BJULIydLxibgV7hvcKGC8WMylFskTudVBA3QMISaVbYOpKsRyqMOHujK8GOrEB%2BBYceWJkR6og5qOcSsCeOTgAsi4goEzwS%2Bdx0TuSNHauuoccZM9JMVledPGMyxjCTipjf36cZ0%2BxMujS%2B2q4Qr1iMDDtAAKfwY7W5DC5ltgViWFfHjg00641r1i2gyYDBk2QzQjyyFNS60JXGsHvN0MwylvU3g8GXJ7OqNfGCEYlZs3cSyo5Ddic4TiwpSY%2Ba%2B1Aemag%2B5GzxiIQVzkXeXVYbuWpzItoxWxp6b%2Fs1NXh&Expires=1709522079\n",
      "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.179.17, 52.92.235.145, 52.92.160.41, ...\n",
      "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.179.17|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 272219353 (260M) [binary/octet-stream]\n",
      "Saving to: ‘ENCFF659UFR.fastq.gz’\n",
      "\n",
      "ENCFF659UFR.fastq.g 100%[===================>] 259.61M  3.72MB/s    in 86s     \n",
      "\n",
      "2024-03-02 18:16:04 (3.01 MB/s) - ‘ENCFF659UFR.fastq.gz’ saved [272219353/272219353]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# ChIP-seq на гистоновой метке\n",
    "!wget https://www.encodeproject.org/files/ENCFF000AXS/@@download/ENCFF000AXS.fastq.gz\n",
    "!wget https://www.encodeproject.org/files/ENCFF000AXZ/@@download/ENCFF000AXZ.fastq.gz\n",
    "# ChIP-seq контроль\n",
    "!wget https://www.encodeproject.org/files/ENCFF659UFR/@@download/ENCFF659UFR.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# распаковка ахрива\n",
    "!gzip -d ENCFF000AXS.fastq.gz ENCFF000AXZ.fastq.gz ENCFF659UFR.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of ENCFF000AXS.fastq\n",
      "Approx 5% complete for ENCFF000AXS.fastq\n",
      "Approx 10% complete for ENCFF000AXS.fastq\n",
      "Approx 15% complete for ENCFF000AXS.fastq\n",
      "Approx 20% complete for ENCFF000AXS.fastq\n",
      "Approx 25% complete for ENCFF000AXS.fastq\n",
      "Approx 30% complete for ENCFF000AXS.fastq\n",
      "Approx 35% complete for ENCFF000AXS.fastq\n",
      "Approx 40% complete for ENCFF000AXS.fastq\n",
      "Approx 45% complete for ENCFF000AXS.fastq\n",
      "Approx 50% complete for ENCFF000AXS.fastq\n",
      "Approx 55% complete for ENCFF000AXS.fastq\n",
      "Approx 60% complete for ENCFF000AXS.fastq\n",
      "Approx 65% complete for ENCFF000AXS.fastq\n",
      "Approx 70% complete for ENCFF000AXS.fastq\n",
      "Approx 75% complete for ENCFF000AXS.fastq\n",
      "Approx 80% complete for ENCFF000AXS.fastq\n",
      "Approx 85% complete for ENCFF000AXS.fastq\n",
      "Approx 90% complete for ENCFF000AXS.fastq\n",
      "Approx 95% complete for ENCFF000AXS.fastq\n",
      "Analysis complete for ENCFF000AXS.fastq\n",
      "Started analysis of ENCFF000AXZ.fastq\n",
      "Approx 5% complete for ENCFF000AXZ.fastq\n",
      "Approx 10% complete for ENCFF000AXZ.fastq\n",
      "Approx 15% complete for ENCFF000AXZ.fastq\n",
      "Approx 20% complete for ENCFF000AXZ.fastq\n",
      "Approx 25% complete for ENCFF000AXZ.fastq\n",
      "Approx 30% complete for ENCFF000AXZ.fastq\n",
      "Approx 35% complete for ENCFF000AXZ.fastq\n",
      "Approx 40% complete for ENCFF000AXZ.fastq\n",
      "Approx 45% complete for ENCFF000AXZ.fastq\n",
      "Approx 50% complete for ENCFF000AXZ.fastq\n",
      "Approx 55% complete for ENCFF000AXZ.fastq\n",
      "Approx 60% complete for ENCFF000AXZ.fastq\n",
      "Approx 65% complete for ENCFF000AXZ.fastq\n",
      "Approx 70% complete for ENCFF000AXZ.fastq\n",
      "Approx 75% complete for ENCFF000AXZ.fastq\n",
      "Approx 80% complete for ENCFF000AXZ.fastq\n",
      "Approx 85% complete for ENCFF000AXZ.fastq\n",
      "Approx 90% complete for ENCFF000AXZ.fastq\n",
      "Approx 95% complete for ENCFF000AXZ.fastq\n",
      "Analysis complete for ENCFF000AXZ.fastq\n",
      "Started analysis of ENCFF659UFR.fastq\n",
      "Approx 5% complete for ENCFF659UFR.fastq\n",
      "Approx 10% complete for ENCFF659UFR.fastq\n",
      "Approx 15% complete for ENCFF659UFR.fastq\n",
      "Approx 20% complete for ENCFF659UFR.fastq\n",
      "Approx 25% complete for ENCFF659UFR.fastq\n",
      "Approx 30% complete for ENCFF659UFR.fastq\n",
      "Approx 35% complete for ENCFF659UFR.fastq\n",
      "Approx 40% complete for ENCFF659UFR.fastq\n",
      "Approx 45% complete for ENCFF659UFR.fastq\n",
      "Approx 50% complete for ENCFF659UFR.fastq\n",
      "Approx 55% complete for ENCFF659UFR.fastq\n",
      "Approx 60% complete for ENCFF659UFR.fastq\n",
      "Approx 65% complete for ENCFF659UFR.fastq\n",
      "Approx 70% complete for ENCFF659UFR.fastq\n",
      "Approx 75% complete for ENCFF659UFR.fastq\n",
      "Approx 80% complete for ENCFF659UFR.fastq\n",
      "Approx 85% complete for ENCFF659UFR.fastq\n",
      "Approx 90% complete for ENCFF659UFR.fastq\n",
      "Approx 95% complete for ENCFF659UFR.fastq\n",
      "Analysis complete for ENCFF659UFR.fastq\n"
     ]
    }
   ],
   "source": [
    "!./FastQC/fastqc ENCFF000AXS.fastq\n",
    "!./FastQC/fastqc ENCFF000AXZ.fastq\n",
    "!./FastQC/fastqc ENCFF659UFR.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Обрезка"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrimmomaticSE: Started with arguments:\n",
      " -phred33 ENCFF000AXS.fastq ENCFF000AXS_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n",
      "Automatically using 1 threads\n",
      "java.io.FileNotFoundException: /mnt/c/Users/User/projects/hse/bioinf/hse_hw2_chip/TruSeq3-SE (No such file or directory)\n",
      "\tat java.base/java.io.FileInputStream.open0(Native Method)\n",
      "\tat java.base/java.io.FileInputStream.open(FileInputStream.java:219)\n",
      "\tat java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)\n",
      "\tat org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)\n",
      "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:110)\n",
      "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:71)\n",
      "\tat org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:32)\n",
      "\tat org.usadellab.trimmomatic.Trimmomatic.createTrimmers(Trimmomatic.java:59)\n",
      "\tat org.usadellab.trimmomatic.TrimmomaticSE.run(TrimmomaticSE.java:318)\n",
      "\tat org.usadellab.trimmomatic.Trimmomatic.main(Trimmomatic.java:85)\n",
      "Input Reads: 7990744 Surviving: 2854768 (35.73%) Dropped: 5135976 (64.27%)\n",
      "TrimmomaticSE: Completed successfully\n"
     ]
    }
   ],
   "source": [
    "!trimmomatic SE -phred33 ENCFF000AXS.fastq ENCFF000AXS_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 \\\n",
    "  LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrimmomaticSE: Started with arguments:\n",
      " -phred33 ENCFF000AXZ.fastq ENCFF000AXZ_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n",
      "Automatically using 1 threads\n",
      "java.io.FileNotFoundException: /mnt/c/Users/User/projects/hse/bioinf/hse_hw2_chip/TruSeq3-SE (No such file or directory)\n",
      "\tat java.base/java.io.FileInputStream.open0(Native Method)\n",
      "\tat java.base/java.io.FileInputStream.open(FileInputStream.java:219)\n",
      "\tat java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)\n",
      "\tat org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)\n",
      "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:110)\n",
      "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:71)\n",
      "\tat org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:32)\n",
      "\tat org.usadellab.trimmomatic.Trimmomatic.createTrimmers(Trimmomatic.java:59)\n",
      "\tat org.usadellab.trimmomatic.TrimmomaticSE.run(TrimmomaticSE.java:318)\n",
      "\tat org.usadellab.trimmomatic.Trimmomatic.main(Trimmomatic.java:85)\n",
      "Input Reads: 16475284 Surviving: 12883109 (78.20%) Dropped: 3592175 (21.80%)\n",
      "TrimmomaticSE: Completed successfully\n"
     ]
    }
   ],
   "source": [
    "!trimmomatic SE -phred33 ENCFF000AXZ.fastq ENCFF000AXZ_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 \\\n",
    "  LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrimmomaticSE: Started with arguments:\n",
      " -phred33 ENCFF659UFR.fastq ENCFF659UFR_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n",
      "Automatically using 1 threads\n",
      "java.io.FileNotFoundException: /mnt/c/Users/User/projects/hse/bioinf/hse_hw2_chip/TruSeq3-SE (No such file or directory)\n",
      "\tat java.base/java.io.FileInputStream.open0(Native Method)\n",
      "\tat java.base/java.io.FileInputStream.open(FileInputStream.java:219)\n",
      "\tat java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)\n",
      "\tat org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)\n",
      "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:110)\n",
      "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:71)\n",
      "\tat org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:32)\n",
      "\tat org.usadellab.trimmomatic.Trimmomatic.createTrimmers(Trimmomatic.java:59)\n",
      "\tat org.usadellab.trimmomatic.TrimmomaticSE.run(TrimmomaticSE.java:318)\n",
      "\tat org.usadellab.trimmomatic.Trimmomatic.main(Trimmomatic.java:85)\n",
      "Input Reads: 7411194 Surviving: 6279078 (84.72%) Dropped: 1132116 (15.28%)\n",
      "TrimmomaticSE: Completed successfully\n"
     ]
    }
   ],
   "source": [
    "!trimmomatic SE -phred33 ENCFF659UFR.fastq ENCFF659UFR_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 \\\n",
    "  LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of ENCFF000AXS_trimmed.fastq\n",
      "Approx 5% complete for ENCFF000AXS_trimmed.fastq\n",
      "Approx 10% complete for ENCFF000AXS_trimmed.fastq\n",
      "Approx 15% complete for ENCFF000AXS_trimmed.fastq\n",
      "Approx 20% complete for ENCFF000AXS_trimmed.fastq\n",
      "Approx 25% complete for ENCFF000AXS_trimmed.fastq\n",
      "Approx 30% complete for ENCFF000AXS_trimmed.fastq\n",
      "Approx 35% complete for ENCFF000AXS_trimmed.fastq\n",
      "Approx 40% complete for ENCFF000AXS_trimmed.fastq\n",
      "Approx 45% complete for ENCFF000AXS_trimmed.fastq\n",
      "Approx 50% complete for ENCFF000AXS_trimmed.fastq\n",
      "Approx 55% complete for ENCFF000AXS_trimmed.fastq\n",
      "Approx 60% complete for ENCFF000AXS_trimmed.fastq\n",
      "Approx 65% complete for ENCFF000AXS_trimmed.fastq\n",
      "Approx 70% complete for ENCFF000AXS_trimmed.fastq\n",
      "Approx 75% complete for ENCFF000AXS_trimmed.fastq\n",
      "Approx 80% complete for ENCFF000AXS_trimmed.fastq\n",
      "Approx 85% complete for ENCFF000AXS_trimmed.fastq\n",
      "Approx 90% complete for ENCFF000AXS_trimmed.fastq\n",
      "Approx 95% complete for ENCFF000AXS_trimmed.fastq\n",
      "Analysis complete for ENCFF000AXS_trimmed.fastq\n",
      "Started analysis of ENCFF000AXZ_trimmed.fastq\n",
      "Approx 5% complete for ENCFF000AXZ_trimmed.fastq\n",
      "Approx 10% complete for ENCFF000AXZ_trimmed.fastq\n",
      "Approx 15% complete for ENCFF000AXZ_trimmed.fastq\n",
      "Approx 20% complete for ENCFF000AXZ_trimmed.fastq\n",
      "Approx 25% complete for ENCFF000AXZ_trimmed.fastq\n",
      "Approx 30% complete for ENCFF000AXZ_trimmed.fastq\n",
      "Approx 35% complete for ENCFF000AXZ_trimmed.fastq\n",
      "Approx 40% complete for ENCFF000AXZ_trimmed.fastq\n",
      "Approx 45% complete for ENCFF000AXZ_trimmed.fastq\n",
      "Approx 50% complete for ENCFF000AXZ_trimmed.fastq\n",
      "Approx 55% complete for ENCFF000AXZ_trimmed.fastq\n",
      "Approx 60% complete for ENCFF000AXZ_trimmed.fastq\n",
      "Approx 65% complete for ENCFF000AXZ_trimmed.fastq\n",
      "Approx 70% complete for ENCFF000AXZ_trimmed.fastq\n",
      "Approx 75% complete for ENCFF000AXZ_trimmed.fastq\n",
      "Approx 80% complete for ENCFF000AXZ_trimmed.fastq\n",
      "Approx 85% complete for ENCFF000AXZ_trimmed.fastq\n",
      "Approx 90% complete for ENCFF000AXZ_trimmed.fastq\n",
      "Approx 95% complete for ENCFF000AXZ_trimmed.fastq\n",
      "Analysis complete for ENCFF000AXZ_trimmed.fastq\n",
      "Started analysis of ENCFF659UFR_trimmed.fastq\n",
      "Approx 5% complete for ENCFF659UFR_trimmed.fastq\n",
      "Approx 10% complete for ENCFF659UFR_trimmed.fastq\n",
      "Approx 15% complete for ENCFF659UFR_trimmed.fastq\n",
      "Approx 20% complete for ENCFF659UFR_trimmed.fastq\n",
      "Approx 25% complete for ENCFF659UFR_trimmed.fastq\n",
      "Approx 30% complete for ENCFF659UFR_trimmed.fastq\n",
      "Approx 35% complete for ENCFF659UFR_trimmed.fastq\n",
      "Approx 40% complete for ENCFF659UFR_trimmed.fastq\n",
      "Approx 45% complete for ENCFF659UFR_trimmed.fastq\n",
      "Approx 50% complete for ENCFF659UFR_trimmed.fastq\n",
      "Approx 55% complete for ENCFF659UFR_trimmed.fastq\n",
      "Approx 60% complete for ENCFF659UFR_trimmed.fastq\n",
      "Approx 65% complete for ENCFF659UFR_trimmed.fastq\n",
      "Approx 70% complete for ENCFF659UFR_trimmed.fastq\n",
      "Approx 75% complete for ENCFF659UFR_trimmed.fastq\n",
      "Approx 80% complete for ENCFF659UFR_trimmed.fastq\n",
      "Approx 85% complete for ENCFF659UFR_trimmed.fastq\n",
      "Approx 90% complete for ENCFF659UFR_trimmed.fastq\n",
      "Approx 95% complete for ENCFF659UFR_trimmed.fastq\n",
      "Analysis complete for ENCFF659UFR_trimmed.fastq\n"
     ]
    }
   ],
   "source": [
    "!./FastQC/fastqc ENCFF000AXS_trimmed.fastq\n",
    "!./FastQC/fastqc ENCFF000AXZ_trimmed.fastq\n",
    "!./FastQC/fastqc ENCFF659UFR_trimmed.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Выравнивание на хромосому"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2024-03-02 18:57:05--  https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr14.fa.gz\n",
      "Resolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.198.53\n",
      "Connecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.198.53|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 29295890 (28M) [application/x-gzip]\n",
      "Saving to: ‘chr14.fa.gz’\n",
      "\n",
      "chr14.fa.gz         100%[===================>]  27.94M  8.65MB/s    in 3.2s    \n",
      "\n",
      "2024-03-02 18:57:09 (8.65 MB/s) - ‘chr14.fa.gz’ saved [29295890/29295890]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr14.fa.gz\n",
    "!gzip -d chr14.fa.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Settings:\n",
      "  Output files: \"chromosome_index.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  chr14.fa\n",
      "Building a SMALL index\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:01\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:00\n",
      "bmax according to bmaxDivN setting: 22642037\n",
      "Using parameters --bmax 16981528 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 16981528 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:00\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:01\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:00\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 1, merged 5; iterating...\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 1.1321e+07 (target: 16981527)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 8\n",
      "  Reserving size (16981528) for bucket 1\n",
      "  Calculating Z arrays for bucket 1\n",
      "  Entering block accumulator loop for bucket 1:\n",
      "  bucket 1: 10%\n",
      "  bucket 1: 20%\n",
      "  bucket 1: 30%\n",
      "  bucket 1: 40%\n",
      "  bucket 1: 50%\n",
      "  bucket 1: 60%\n",
      "  bucket 1: 70%\n",
      "  bucket 1: 80%\n",
      "  bucket 1: 90%\n",
      "  bucket 1: 100%\n",
      "  Sorting block of length 10995868 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:01\n",
      "Returning block of 10995869 for bucket 1\n",
      "Getting block 2 of 8\n",
      "  Reserving size (16981528) for bucket 2\n",
      "  Calculating Z arrays for bucket 2\n",
      "  Entering block accumulator loop for bucket 2:\n",
      "  bucket 2: 10%\n",
      "  bucket 2: 20%\n",
      "  bucket 2: 30%\n",
      "  bucket 2: 40%\n",
      "  bucket 2: 50%\n",
      "  bucket 2: 60%\n",
      "  bucket 2: 70%\n",
      "  bucket 2: 80%\n",
      "  bucket 2: 90%\n",
      "  bucket 2: 100%\n",
      "  Sorting block of length 10732981 for bucket 2\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:02\n",
      "Returning block of 10732982 for bucket 2\n",
      "Getting block 3 of 8\n",
      "  Reserving size (16981528) for bucket 3\n",
      "  Calculating Z arrays for bucket 3\n",
      "  Entering block accumulator loop for bucket 3:\n",
      "  bucket 3: 10%\n",
      "  bucket 3: 20%\n",
      "  bucket 3: 30%\n",
      "  bucket 3: 40%\n",
      "  bucket 3: 50%\n",
      "  bucket 3: 60%\n",
      "  bucket 3: 70%\n",
      "  bucket 3: 80%\n",
      "  bucket 3: 90%\n",
      "  bucket 3: 100%\n",
      "  Sorting block of length 14209021 for bucket 3\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:03\n",
      "Returning block of 14209022 for bucket 3\n",
      "Getting block 4 of 8\n",
      "  Reserving size (16981528) for bucket 4\n",
      "  Calculating Z arrays for bucket 4\n",
      "  Entering block accumulator loop for bucket 4:\n",
      "  bucket 4: 10%\n",
      "  bucket 4: 20%\n",
      "  bucket 4: 30%\n",
      "  bucket 4: 40%\n",
      "  bucket 4: 50%\n",
      "  bucket 4: 60%\n",
      "  bucket 4: 70%\n",
      "  bucket 4: 80%\n",
      "  bucket 4: 90%\n",
      "  bucket 4: 100%\n",
      "  Sorting block of length 16396496 for bucket 4\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:02\n",
      "Returning block of 16396497 for bucket 4\n",
      "Getting block 5 of 8\n",
      "  Reserving size (16981528) for bucket 5\n",
      "  Calculating Z arrays for bucket 5\n",
      "  Entering block accumulator loop for bucket 5:\n",
      "  bucket 5: 10%\n",
      "  bucket 5: 20%\n",
      "  bucket 5: 30%\n",
      "  bucket 5: 40%\n",
      "  bucket 5: 50%\n",
      "  bucket 5: 60%\n",
      "  bucket 5: 70%\n",
      "  bucket 5: 80%\n",
      "  bucket 5: 90%\n",
      "  bucket 5: 100%\n",
      "  Sorting block of length 6586600 for bucket 5\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:01\n",
      "Returning block of 6586601 for bucket 5\n",
      "Getting block 6 of 8\n",
      "  Reserving size (16981528) for bucket 6\n",
      "  Calculating Z arrays for bucket 6\n",
      "  Entering block accumulator loop for bucket 6:\n",
      "  bucket 6: 10%\n",
      "  bucket 6: 20%\n",
      "  bucket 6: 30%\n",
      "  bucket 6: 40%\n",
      "  bucket 6: 50%\n",
      "  bucket 6: 60%\n",
      "  bucket 6: 70%\n",
      "  bucket 6: 80%\n",
      "  bucket 6: 90%\n",
      "  bucket 6: 100%\n",
      "  Sorting block of length 14286880 for bucket 6\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:02\n",
      "Returning block of 14286881 for bucket 6\n",
      "Getting block 7 of 8\n",
      "  Reserving size (16981528) for bucket 7\n",
      "  Calculating Z arrays for bucket 7\n",
      "  Entering block accumulator loop for bucket 7:\n",
      "  bucket 7: 10%\n",
      "  bucket 7: 20%\n",
      "  bucket 7: 30%\n",
      "  bucket 7: 40%\n",
      "  bucket 7: 50%\n",
      "  bucket 7: 60%\n",
      "  bucket 7: 70%\n",
      "  bucket 7: 80%\n",
      "  bucket 7: 90%\n",
      "  bucket 7: 100%\n",
      "  Sorting block of length 16145458 for bucket 7\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:02\n",
      "Returning block of 16145459 for bucket 7\n",
      "Getting block 8 of 8\n",
      "  Reserving size (16981528) for bucket 8\n",
      "  Calculating Z arrays for bucket 8\n",
      "  Entering block accumulator loop for bucket 8:\n",
      "  bucket 8: 10%\n",
      "  bucket 8: 20%\n",
      "  bucket 8: 30%\n",
      "  bucket 8: 40%\n",
      "  bucket 8: 50%\n",
      "  bucket 8: 60%\n",
      "  bucket 8: 70%\n",
      "  bucket 8: 80%\n",
      "  bucket 8: 90%\n",
      "  bucket 8: 100%\n",
      "  Sorting block of length 1214838 for bucket 8\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 1214839 for bucket 8\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 26673415\n",
      "fchr[G]: 45097173\n",
      "fchr[T]: 63656206\n",
      "fchr[$]: 90568149\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 34384187 bytes to primary EBWT file: chromosome_index.1.bt2\n",
      "Wrote 22642044 bytes to secondary EBWT file: chromosome_index.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 90568149\n",
      "    bwtLen: 90568150\n",
      "    sz: 22642038\n",
      "    bwtSz: 22642038\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 5660510\n",
      "    offsSz: 22642040\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 471710\n",
      "    numLines: 471710\n",
      "    ebwtTotLen: 30189440\n",
      "    ebwtTotSz: 30189440\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:00:38\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:01\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:01\n",
      "  Time to reverse reference sequence: 00:00:00\n",
      "bmax according to bmaxDivN setting: 22642037\n",
      "Using parameters --bmax 16981528 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 16981528 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:01\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:00\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:01\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 1, merged 7; iterating...\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 1, merged 0; iterating...\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 1.29383e+07 (target: 16981527)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 7\n",
      "  Reserving size (16981528) for bucket 1\n",
      "  Calculating Z arrays for bucket 1\n",
      "  Entering block accumulator loop for bucket 1:\n",
      "  bucket 1: 10%\n",
      "  bucket 1: 20%\n",
      "  bucket 1: 30%\n",
      "  bucket 1: 40%\n",
      "  bucket 1: 50%\n",
      "  bucket 1: 60%\n",
      "  bucket 1: 70%\n",
      "  bucket 1: 80%\n",
      "  bucket 1: 90%\n",
      "  bucket 1: 100%\n",
      "  Sorting block of length 12277527 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:01\n",
      "Returning block of 12277528 for bucket 1\n",
      "Getting block 2 of 7\n",
      "  Reserving size (16981528) for bucket 2\n",
      "  Calculating Z arrays for bucket 2\n",
      "  Entering block accumulator loop for bucket 2:\n",
      "  bucket 2: 10%\n",
      "  bucket 2: 20%\n",
      "  bucket 2: 30%\n",
      "  bucket 2: 40%\n",
      "  bucket 2: 50%\n",
      "  bucket 2: 60%\n",
      "  bucket 2: 70%\n",
      "  bucket 2: 80%\n",
      "  bucket 2: 90%\n",
      "  bucket 2: 100%\n",
      "  Sorting block of length 16458020 for bucket 2\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:02\n",
      "Returning block of 16458021 for bucket 2\n",
      "Getting block 3 of 7\n",
      "  Reserving size (16981528) for bucket 3\n",
      "  Calculating Z arrays for bucket 3\n",
      "  Entering block accumulator loop for bucket 3:\n",
      "  bucket 3: 10%\n",
      "  bucket 3: 20%\n",
      "  bucket 3: 30%\n",
      "  bucket 3: 40%\n",
      "  bucket 3: 50%\n",
      "  bucket 3: 60%\n",
      "  bucket 3: 70%\n",
      "  bucket 3: 80%\n",
      "  bucket 3: 90%\n",
      "  bucket 3: 100%\n",
      "  Sorting block of length 15884098 for bucket 3\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:03\n",
      "Returning block of 15884099 for bucket 3\n",
      "Getting block 4 of 7\n",
      "  Reserving size (16981528) for bucket 4\n",
      "  Calculating Z arrays for bucket 4\n",
      "  Entering block accumulator loop for bucket 4:\n",
      "  bucket 4: 10%\n",
      "  bucket 4: 20%\n",
      "  bucket 4: 30%\n",
      "  bucket 4: 40%\n",
      "  bucket 4: 50%\n",
      "  bucket 4: 60%\n",
      "  bucket 4: 70%\n",
      "  bucket 4: 80%\n",
      "  bucket 4: 90%\n",
      "  bucket 4: 100%\n",
      "  Sorting block of length 10967214 for bucket 4\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:01\n",
      "Returning block of 10967215 for bucket 4\n",
      "Getting block 5 of 7\n",
      "  Reserving size (16981528) for bucket 5\n",
      "  Calculating Z arrays for bucket 5\n",
      "  Entering block accumulator loop for bucket 5:\n",
      "  bucket 5: 10%\n",
      "  bucket 5: 20%\n",
      "  bucket 5: 30%\n",
      "  bucket 5: 40%\n",
      "  bucket 5: 50%\n",
      "  bucket 5: 60%\n",
      "  bucket 5: 70%\n",
      "  bucket 5: 80%\n",
      "  bucket 5: 90%\n",
      "  bucket 5: 100%\n",
      "  Sorting block of length 10048387 for bucket 5\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:01\n",
      "Returning block of 10048388 for bucket 5\n",
      "Getting block 6 of 7\n",
      "  Reserving size (16981528) for bucket 6\n",
      "  Calculating Z arrays for bucket 6\n",
      "  Entering block accumulator loop for bucket 6:\n",
      "  bucket 6: 10%\n",
      "  bucket 6: 20%\n",
      "  bucket 6: 30%\n",
      "  bucket 6: 40%\n",
      "  bucket 6: 50%\n",
      "  bucket 6: 60%\n",
      "  bucket 6: 70%\n",
      "  bucket 6: 80%\n",
      "  bucket 6: 90%\n",
      "  bucket 6: 100%\n",
      "  Sorting block of length 11270887 for bucket 6\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:02\n",
      "Returning block of 11270888 for bucket 6\n",
      "Getting block 7 of 7\n",
      "  Reserving size (16981528) for bucket 7\n",
      "  Calculating Z arrays for bucket 7\n",
      "  Entering block accumulator loop for bucket 7:\n",
      "  bucket 7: 10%\n",
      "  bucket 7: 20%\n",
      "  bucket 7: 30%\n",
      "  bucket 7: 40%\n",
      "  bucket 7: 50%\n",
      "  bucket 7: 60%\n",
      "  bucket 7: 70%\n",
      "  bucket 7: 80%\n",
      "  bucket 7: 90%\n",
      "  bucket 7: 100%\n",
      "  Sorting block of length 13662010 for bucket 7\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:02\n",
      "Returning block of 13662011 for bucket 7\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 26673415\n",
      "fchr[G]: 45097173\n",
      "fchr[T]: 63656206\n",
      "fchr[$]: 90568149\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 34384187 bytes to primary EBWT file: chromosome_index.rev.1.bt2\n",
      "Wrote 22642044 bytes to secondary EBWT file: chromosome_index.rev.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 90568149\n",
      "    bwtLen: 90568150\n",
      "    sz: 22642038\n",
      "    bwtSz: 22642038\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 5660510\n",
      "    offsSz: 22642040\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 471710\n",
      "    numLines: 471710\n",
      "    ebwtTotLen: 30189440\n",
      "    ebwtTotSz: 30189440\n",
      "    color: 0\n",
      "    reverse: 1\n",
      "Total time for backward call to driver() for mirror index: 00:00:39\n",
      "CPU times: user 955 ms, sys: 359 ms, total: 1.31 s\n",
      "Wall time: 1min 17s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!bowtie2-build chr14.fa chromosome_index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir bowtie2_res"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7990744 reads; of these:\n",
      "  7990744 (100.00%) were unpaired; of these:\n",
      "    6814930 (85.29%) aligned 0 times\n",
      "    315305 (3.95%) aligned exactly 1 time\n",
      "    860509 (10.77%) aligned >1 times\n",
      "14.71% overall alignment rate\n",
      "CPU times: user 1.32 s, sys: 406 ms, total: 1.72 s\n",
      "Wall time: 1min 54s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!bowtie2  -p 2 \\\n",
    "          -x chromosome_index \\\n",
    "          -U ENCFF000AXS.fastq \\\n",
    "          -S bowtie2_res/res_AXS.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16475284 reads; of these:\n",
      "  16475284 (100.00%) were unpaired; of these:\n",
      "    13940229 (84.61%) aligned 0 times\n",
      "    643213 (3.90%) aligned exactly 1 time\n",
      "    1891842 (11.48%) aligned >1 times\n",
      "15.39% overall alignment rate\n",
      "CPU times: user 2.62 s, sys: 1.02 s, total: 3.64 s\n",
      "Wall time: 4min 1s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!bowtie2  -p 2 \\\n",
    "          -x chromosome_index \\\n",
    "          -U ENCFF000AXZ.fastq \\\n",
    "          -S bowtie2_res/res_AXZ.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7411194 reads; of these:\n",
      "  7411194 (100.00%) were unpaired; of these:\n",
      "    6163112 (83.16%) aligned 0 times\n",
      "    311390 (4.20%) aligned exactly 1 time\n",
      "    936692 (12.64%) aligned >1 times\n",
      "16.84% overall alignment rate\n",
      "CPU times: user 1.46 s, sys: 497 ms, total: 1.96 s\n",
      "Wall time: 2min 4s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!bowtie2  -p 2 \\\n",
    "          -x chromosome_index \\\n",
    "          -U ENCFF659UFR.fastq \\\n",
    "          -S bowtie2_res/res_UFR.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Общее число ридов</th>\n",
       "      <th>Выровнившиеся уникально</th>\n",
       "      <th>Выровнившиеся уникально (%)</th>\n",
       "      <th>Выровнившиеся неуникально</th>\n",
       "      <th>Выровнившиеся неуникально (%)</th>\n",
       "      <th>Не выровнившиеся</th>\n",
       "      <th>Не выровнившиеся (%)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENCFF000AXS</th>\n",
       "      <td>7990744</td>\n",
       "      <td>315305</td>\n",
       "      <td>3.95%</td>\n",
       "      <td>860509</td>\n",
       "      <td>10.77%</td>\n",
       "      <td>6814930</td>\n",
       "      <td>85.29%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENCFF000AXZ</th>\n",
       "      <td>16475284</td>\n",
       "      <td>643213</td>\n",
       "      <td>3.90%</td>\n",
       "      <td>1891842</td>\n",
       "      <td>11.48%</td>\n",
       "      <td>13940229</td>\n",
       "      <td>84.61%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENCFF659UFR</th>\n",
       "      <td>7411194</td>\n",
       "      <td>311390</td>\n",
       "      <td>4.20%</td>\n",
       "      <td>936692</td>\n",
       "      <td>12.64%</td>\n",
       "      <td>6163112</td>\n",
       "      <td>83.16%</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Общее число ридов Выровнившиеся уникально  \\\n",
       "ENCFF000AXS            7990744                 315305    \n",
       "ENCFF000AXZ           16475284                 643213    \n",
       "ENCFF659UFR            7411194                 311390    \n",
       "\n",
       "            Выровнившиеся уникально (%) Выровнившиеся неуникально  \\\n",
       "ENCFF000AXS                       3.95%                   860509    \n",
       "ENCFF000AXZ                       3.90%                  1891842    \n",
       "ENCFF659UFR                       4.20%                   936692    \n",
       "\n",
       "            Выровнившиеся неуникально (%) Не выровнившиеся  \\\n",
       "ENCFF000AXS                        10.77%          6814930   \n",
       "ENCFF000AXZ                        11.48%         13940229   \n",
       "ENCFF659UFR                        12.64%         6163112    \n",
       "\n",
       "            Не выровнившиеся (%)  \n",
       "ENCFF000AXS               85.29%  \n",
       "ENCFF000AXZ               84.61%  \n",
       "ENCFF659UFR               83.16%  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "data = {'Общее число ридов':[7990744,\n",
    "                             16475284,\n",
    "                             7411194],\n",
    "        'Выровнившиеся уникально':['315305 ',\n",
    "                                   '643213 ',\n",
    "                                   '311390 '],\n",
    "        'Выровнившиеся уникально (%)':['3.95%', \n",
    "                                       '3.90%',\n",
    "                                       '4.20%'],\n",
    "        'Выровнившиеся неуникально':['860509 ',\n",
    "                                     '1891842 ',\n",
    "                                     '936692 '],\n",
    "        'Выровнившиеся неуникально (%)':['10.77%',\n",
    "                                         '11.48%',\n",
    "                                         '12.64%'],\n",
    "        'Не выровнившиеся':['6814930',\n",
    "                            '13940229',\n",
    "                            '6163112 '],\n",
    "        'Не выровнившиеся (%)':['85.29%',\n",
    "                                '84.61%',\n",
    "                                '83.16%']}\n",
    "\n",
    "df = pd.DataFrame(data, index =['ENCFF000AXS',\n",
    "                                'ENCFF000AXZ',\n",
    "                                'ENCFF659UFR'])\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Процент выравниваний получился именно таким из-за следующих факторов:\n",
    "\n",
    "Качество последовательностей чтения: Качество последовательностей чтения влияет на способность Bowtie выравнивать их с геномом. Если в последовательностях чтения много ошибок, Bowtie может не найти точных совпадений и, следовательно, процент выравниваний будет ниже.\n",
    "\n",
    "Строгость параметров выравнивания: Строгость параметров выравнивания, таких как количество разрешенных несоответствий и разрывов, влияет на процент выравниваний. Более строгие параметры приводят к меньшему количеству выравниваний, в то время как менее строгие параметры приводят к большему количеству выравниваний.\n",
    "\n",
    "Размер и сложность генома: Размер и сложность генома также влияют на процент выравниваний. Более крупные и сложные геномы содержат больше повторов и других областей с высокой степенью сходства, что может затруднить выравнивание последовательностей чтения.\n",
    "\n",
    "Алгоритм выравнивания: Алгоритм выравнивания, используемый Bowtie, также влияет на процент выравниваний. Разные алгоритмы имеют разную эффективность и точность, что может привести к разным процентам выравниваний.\n",
    "\n",
    "В приведенных примерах процент выравниваний различается из-за различий в качестве последовательностей чтения, строгости параметров выравнивания и размера и сложности генома. В первом примере процент выравниваний ниже, вероятно, из-за более строгого набора параметров выравнивания или более сложного генома. Во втором примере процент выравниваний выше, вероятно, из-за менее строгого набора параметров выравнивания или более простого генома."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "!samtools view -h -S -b -o bowtie2_res/res_AXS.bam bowtie2_res/res_AXS.sam\n",
    "!samtools view -h -S -b -o bowtie2_res/res_AXZ.bam bowtie2_res/res_AXZ.sam\n",
    "!samtools view -h -S -b -o bowtie2_res/res_UFR.bam bowtie2_res/res_UFR.sam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Peak calling"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘macs2’: File exists\n"
     ]
    }
   ],
   "source": [
    "!mkdir macs2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO  @ Sat, 02 Mar 2024 21:05:22: \n",
      "# Command line: callpeak --broad -t bowtie2_res/res_AXS.sam -c bowtie2_res/res_UFR.sam -f SAM --outdir macs2 -n ENCFF000AXS\n",
      "# ARGUMENTS LIST:\n",
      "# name = ENCFF000AXS\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['bowtie2_res/res_AXS.sam']\n",
      "# control file = ['bowtie2_res/res_UFR.sam']\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
      "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Sat, 02 Mar 2024 21:05:22: #1 read tag files... \n",
      "INFO  @ Sat, 02 Mar 2024 21:05:22: #1 read treatment tags... \n",
      "INFO  @ Sat, 02 Mar 2024 21:05:49:  6000000 \n",
      "INFO  @ Sat, 02 Mar 2024 21:05:58: #1.2 read input tags... \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:26:  6000000 \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:32: #1 tag size is determined as 36 bps \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:32: #1 tag size = 36.0 \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:32: #1  total tags in treatment: 1175814 \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:32: #1 user defined the maximum tags... \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:32: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:32: #1  tags after filtering in treatment: 1073805 \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:32: #1  Redundant rate of treatment: 0.09 \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:32: #1  total tags in control: 1248082 \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:32: #1 user defined the maximum tags... \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:32: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:32: #1  tags after filtering in control: 1144343 \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:32: #1  Redundant rate of control: 0.08 \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:32: #1 finished! \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:32: #2 Build Peak Model... \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:32: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:33: #2 number of paired peaks: 11256 \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:33: start model_add_line... \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:33: start X-correlation... \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:33: end of X-cor \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:33: #2 finished! \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:33: #2 predicted fragment length is 37 bps \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:33: #2 alternative fragment length(s) may be 37,100,475 bps \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:33: #2.2 Generate R script for model : macs2/ENCFF000AXS_model.r \n",
      "WARNING @ Sat, 02 Mar 2024 21:06:33: #2 Since the d (37) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
      "WARNING @ Sat, 02 Mar 2024 21:06:33: #2 You may need to consider one of the other alternative d(s): 37,100,475 \n",
      "WARNING @ Sat, 02 Mar 2024 21:06:33: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:33: #3 Call peaks... \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:33: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:33: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:36: #3 Call peaks for each chromosome... \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:37: #4 Write output xls file... macs2/ENCFF000AXS_peaks.xls \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:37: #4 Write broad peak in broadPeak format file... macs2/ENCFF000AXS_peaks.broadPeak \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:37: #4 Write broad peak in bed12/gappedPeak format file... macs2/ENCFF000AXS_peaks.gappedPeak \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:37: Done! \n"
     ]
    }
   ],
   "source": [
    "!macs2 callpeak --broad -t bowtie2_res/res_AXS.sam \\\n",
    "    -c bowtie2_res/res_UFR.sam \\\n",
    " \t  -f SAM \\\n",
    "\t  --outdir macs2 \\\n",
    "\t  -n ENCFF000AXS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO  @ Sat, 02 Mar 2024 21:06:38: \n",
      "# Command line: callpeak --broad -t bowtie2_res/res_AXZ.sam -c bowtie2_res/res_UFR.sam -f SAM --outdir macs2 -n ENCFF000AXZ\n",
      "# ARGUMENTS LIST:\n",
      "# name = ENCFF000AXZ\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['bowtie2_res/res_AXZ.sam']\n",
      "# control file = ['bowtie2_res/res_UFR.sam']\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
      "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:38: #1 read tag files... \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:38: #1 read treatment tags... \n",
      "INFO  @ Sat, 02 Mar 2024 21:06:49:  2000000 \n",
      "INFO  @ Sat, 02 Mar 2024 21:07:18:  7000000 \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:13: #1.2 read input tags... \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:40:  6000000 \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: #1 tag size is determined as 36 bps \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: #1 tag size = 36.0 \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: #1  total tags in treatment: 2535055 \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: #1 user defined the maximum tags... \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: #1  tags after filtering in treatment: 2201425 \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: #1  Redundant rate of treatment: 0.13 \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: #1  total tags in control: 1248082 \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: #1 user defined the maximum tags... \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: #1  tags after filtering in control: 1144343 \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: #1  Redundant rate of control: 0.08 \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: #1 finished! \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: #2 Build Peak Model... \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: #2 number of paired peaks: 23168 \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: start model_add_line... \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: start X-correlation... \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: end of X-cor \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: #2 finished! \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: #2 predicted fragment length is 36 bps \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: #2 alternative fragment length(s) may be 36 bps \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: #2.2 Generate R script for model : macs2/ENCFF000AXZ_model.r \n",
      "WARNING @ Sat, 02 Mar 2024 21:08:47: #2 Since the d (36) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
      "WARNING @ Sat, 02 Mar 2024 21:08:47: #2 You may need to consider one of the other alternative d(s): 36 \n",
      "WARNING @ Sat, 02 Mar 2024 21:08:47: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: #3 Call peaks... \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:47: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:51: #3 Call peaks for each chromosome... \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:53: #4 Write output xls file... macs2/ENCFF000AXZ_peaks.xls \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:53: #4 Write broad peak in broadPeak format file... macs2/ENCFF000AXZ_peaks.broadPeak \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:53: #4 Write broad peak in bed12/gappedPeak format file... macs2/ENCFF000AXZ_peaks.gappedPeak \n",
      "INFO  @ Sat, 02 Mar 2024 21:08:53: Done! \n"
     ]
    }
   ],
   "source": [
    "!macs2 callpeak --broad -t bowtie2_res/res_AXZ.sam \\\n",
    "    -c bowtie2_res/res_UFR.sam \\\n",
    " \t  -f SAM \\\n",
    "\t  --outdir macs2 \\\n",
    "\t  -n ENCFF000AXZ"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2024-03-02 21:26:53--  https://www.encodeproject.org/files/ENCFF118FXI/@@download/ENCFF118FXI.bed.gz\n",
      "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
      "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
      "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
      "Location: https://encode-public.s3.amazonaws.com/2023/01/15/309fde0b-8530-40cb-a56a-7013a33be91c/ENCFF118FXI.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF118FXI.bed.gz&AWSAccessKeyId=ASIATGZNGCNXV6PDTE7V&Signature=wu4jz%2Bn%2FUQcnrRn3FRQoGnOsCOM%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEFMaCXVzLXdlc3QtMiJIMEYCIQDCD9m7pjAdGf8zpT3%2BkPgJ8mLH367PbKLknvj2VUrfhAIhANZUx1tzuvFFjJ77Cy2DAtxB75g4nYMr0GD6Gmg60Ig9KrMFCEwQABoMMjIwNzQ4NzE0ODYzIgzzEdynGvdVKQ%2BofKAqkAUuMU9vcTuWQztX3xu80cEc9U%2FUwUdPEB5f%2F0Vv1aGIWt2LezLqLeN2uU1osapZ164OpXa9EIXgSuPPXaKJG0LVrggq4jXLXKQ6TK2M1oOj48LY%2FEvoq4lSM04Z%2Fm0mtYGMo7m42Cn6MUTpUzDRI7XiArQCtLLrIW0hulaKk8Nr%2BF0HQjJ7LaZFNQ3%2BNkSyTRcHs3BJgaRnyoKOy5ZB1UyPxj1gjh8QpsiIubbo47Vw71P14vcFYiiQ0uF7EJmmNsxIzAU7MHfmonhbxccZkGRL54B6UOyAZRgVTykzkZKC%2BruhYOQ0ORPJF5lj%2Ba%2BA7CpIYrlHR5OsQI%2ByT1%2FGhNhF8bno9BYnpP3dVFmRVCIxbzJ454yyljpB67gfUW%2BNPI00ToiNjncyG8sgEB%2B9kaVrAgxoPtFQ9UEGO3TXF%2FT77HhL7SyxOl9lVtC6Uw62T98zKx7sIkzovtKKmE5GjXuA1F3Vi9Ao5YvJxbs8dnVIcjJG9ZEwVk9A5jT%2BdgZRREhouM6319aU3V9zvw6Di%2BAwg0SguCTnFR7E3JiQQc%2B%2Bt92ttN6Wr8YSoDmlvcXdV4T44zskbGfkk2t1QobyQYpPSUqw53GyHwfHXqIhtp%2ByVrwNsJ7NdETvpbnJePicmHtV43TdUNxfc68ql5IGgjDZP%2Fgh8THxVoEq2LZdxqHDzJOp17XUOgwLUQ7eg12Qew16rmU9ZH8gwF%2FXpRJhFy7sHtaWb4SJjtSSKh9uhQiHv26OLrl01iVwskKzRBEIik%2F%2BgT43tAcTnolXTRRSy02YCBKnQgFVa6BfgEhtIm%2FOg%2FfzZkBkvXPe0o7hi%2B2bkZoJkLFdl%2FOIenhGxIFET4buCnRqM9G2y0yM3bH2GtLS7DCJ3o2vBjqwAeSavLKcFmA0lJ7kHhCAE93m3WNSUUQ8EVb%2FrGP2P6FrVOeIwLfJgGFTdnVzMRMvgjQVSSUm3wp2GMDyx%2BNWnjEVpRzhAQNGgImaEBDjo0anm2xGhDR6HjFDpsJ%2ByTE1JA%2Fiq7nivrQO5x9VOtG4lVG4j3pGs8%2BneocNKVaRt25660zm60q3KHmn12WcZ6QhF5y0Mel5jeOlADqyT6Sf4%2F7pdvp3bugBOdIpXa5stQQz&Expires=1709533616 [following]\n",
      "--2024-03-02 21:26:54--  https://encode-public.s3.amazonaws.com/2023/01/15/309fde0b-8530-40cb-a56a-7013a33be91c/ENCFF118FXI.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF118FXI.bed.gz&AWSAccessKeyId=ASIATGZNGCNXV6PDTE7V&Signature=wu4jz%2Bn%2FUQcnrRn3FRQoGnOsCOM%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEFMaCXVzLXdlc3QtMiJIMEYCIQDCD9m7pjAdGf8zpT3%2BkPgJ8mLH367PbKLknvj2VUrfhAIhANZUx1tzuvFFjJ77Cy2DAtxB75g4nYMr0GD6Gmg60Ig9KrMFCEwQABoMMjIwNzQ4NzE0ODYzIgzzEdynGvdVKQ%2BofKAqkAUuMU9vcTuWQztX3xu80cEc9U%2FUwUdPEB5f%2F0Vv1aGIWt2LezLqLeN2uU1osapZ164OpXa9EIXgSuPPXaKJG0LVrggq4jXLXKQ6TK2M1oOj48LY%2FEvoq4lSM04Z%2Fm0mtYGMo7m42Cn6MUTpUzDRI7XiArQCtLLrIW0hulaKk8Nr%2BF0HQjJ7LaZFNQ3%2BNkSyTRcHs3BJgaRnyoKOy5ZB1UyPxj1gjh8QpsiIubbo47Vw71P14vcFYiiQ0uF7EJmmNsxIzAU7MHfmonhbxccZkGRL54B6UOyAZRgVTykzkZKC%2BruhYOQ0ORPJF5lj%2Ba%2BA7CpIYrlHR5OsQI%2ByT1%2FGhNhF8bno9BYnpP3dVFmRVCIxbzJ454yyljpB67gfUW%2BNPI00ToiNjncyG8sgEB%2B9kaVrAgxoPtFQ9UEGO3TXF%2FT77HhL7SyxOl9lVtC6Uw62T98zKx7sIkzovtKKmE5GjXuA1F3Vi9Ao5YvJxbs8dnVIcjJG9ZEwVk9A5jT%2BdgZRREhouM6319aU3V9zvw6Di%2BAwg0SguCTnFR7E3JiQQc%2B%2Bt92ttN6Wr8YSoDmlvcXdV4T44zskbGfkk2t1QobyQYpPSUqw53GyHwfHXqIhtp%2ByVrwNsJ7NdETvpbnJePicmHtV43TdUNxfc68ql5IGgjDZP%2Fgh8THxVoEq2LZdxqHDzJOp17XUOgwLUQ7eg12Qew16rmU9ZH8gwF%2FXpRJhFy7sHtaWb4SJjtSSKh9uhQiHv26OLrl01iVwskKzRBEIik%2F%2BgT43tAcTnolXTRRSy02YCBKnQgFVa6BfgEhtIm%2FOg%2FfzZkBkvXPe0o7hi%2B2bkZoJkLFdl%2FOIenhGxIFET4buCnRqM9G2y0yM3bH2GtLS7DCJ3o2vBjqwAeSavLKcFmA0lJ7kHhCAE93m3WNSUUQ8EVb%2FrGP2P6FrVOeIwLfJgGFTdnVzMRMvgjQVSSUm3wp2GMDyx%2BNWnjEVpRzhAQNGgImaEBDjo0anm2xGhDR6HjFDpsJ%2ByTE1JA%2Fiq7nivrQO5x9VOtG4lVG4j3pGs8%2BneocNKVaRt25660zm60q3KHmn12WcZ6QhF5y0Mel5jeOlADqyT6Sf4%2F7pdvp3bugBOdIpXa5stQQz&Expires=1709533616\n",
      "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.179.17, 52.92.235.145, 52.92.160.41, ...\n",
      "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.179.17|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 890506 (870K) [binary/octet-stream]\n",
      "Saving to: ‘ENCFF118FXI.bed.gz’\n",
      "\n",
      "ENCFF118FXI.bed.gz  100%[===================>] 869.63K   879KB/s    in 1.0s    \n",
      "\n",
      "2024-03-02 21:26:56 (879 KB/s) - ‘ENCFF118FXI.bed.gz’ saved [890506/890506]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget https://www.encodeproject.org/files/ENCFF118FXI/@@download/ENCFF118FXI.bed.gz\n",
    "!gzip -d ENCFF118FXI.bed.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Сравнение результатов"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir venn1\n",
    "!mkdir venn2\n",
    "!mkdir venn3\n",
    "!mkdir venn4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Generating a 2-way \"venn\" diagram. Please wait...\n",
      "\n",
      "\n",
      "Done! Please check your results @ venn1. \n",
      "Thank you for using Intervene!\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!intervene venn -i macs2/ENCFF000AXS_peaks.broadPeak ENCFF118FXI.bed -o venn1 --filenames"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Generating a 2-way \"venn\" diagram. Please wait...\n",
      "\n",
      "\n",
      "Done! Please check your results @ venn2. \n",
      "Thank you for using Intervene!\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!intervene venn -i ENCFF118FXI.bed macs2/ENCFF000AXS_peaks.broadPeak -o venn2 --filenames"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Generating a 2-way \"venn\" diagram. Please wait...\n",
      "\n",
      "\n",
      "Done! Please check your results @ venn3. \n",
      "Thank you for using Intervene!\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!intervene venn -i macs2/ENCFF000AXZ_peaks.broadPeak ENCFF118FXI.bed -o venn3 --filenames"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Generating a 2-way \"venn\" diagram. Please wait...\n",
      "\n",
      "\n",
      "Done! Please check your results @ venn4. \n",
      "Thank you for using Intervene!\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!intervene venn -i ENCFF118FXI.bed macs2/ENCFF000AXZ_peaks.broadPeak -o venn4 --filenames"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Бонус"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2024-03-02 22:44:55--  https://www.encodeproject.org/files/ENCFF285FHP/@@download/ENCFF285FHP.bam\n",
      "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
      "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
      "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
      "Location: https://encode-public.s3.amazonaws.com/2023/01/15/5787db88-2846-4a48-8eb5-0bafa02771c9/ENCFF285FHP.bam?response-content-disposition=attachment%3B%20filename%3DENCFF285FHP.bam&AWSAccessKeyId=ASIATGZNGCNXY37BXRX4&Signature=rupRwOcUNWGjJM%2FO3TGddjOg%2B2w%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEFQaCXVzLXdlc3QtMiJIMEYCIQC6PdtSmfTx6QZwuEiXIb7ejZ%2F4TBmd8uEYX1Eb1h3sfAIhAOhudH49qg5H6faOR9aV4XeFxBK0ZpMF1Twg0sVw%2BPniKrMFCE0QABoMMjIwNzQ4NzE0ODYzIgzzwbFwvJgJ7AmXaxkqkAWVguA0uwr0o%2FOxIUBbsRpbw9xx7WSRaJ5o0x5c%2BfWUQgYzAS8eEZRZRQQIu7X7yADaXBMotYhTUCQNhUjasW4oT5RApJQ9XR857jMaSZxKskhhfKQP66giGFID5pxs4acJ2Z6A6WuT2oedcANoEG32EPZXLd2UBe66%2Bo9ahtxXkE5J%2B1Fu4IppNWRbnu4knxv7VY0AfnmVh8olWr6jC0Sr6axmm00SXagsbMNVv8z2DXg3x8L6Ha0lJMj7pLjqxxOXW%2ByEzboO9wPFb8jNMSs%2FGcnOAZu7t6eD9CnYZOS3GXGe%2Fn%2BItOqpmiwS%2F6KMx8TmzimZe8P9KjL2poD5SqGhGtL%2F8PcZLoHK3Ea4fiSI5NqJvKtVwNII%2FTQjixU5g7WEwLW8tq%2FODimrNGx3wT6fHdSZ9KBG6uACA1m0iWBAb7W40DkgfP2CKAcl7vYffkQ%2Fp1WxxZKNxeEIOH9nWBsUowjTF0J7g2Ei3mdp2rDvXMu1bgk9bdAS6LO76RB0uUH3xjaZo4RJhomRiaj%2FRTraN3WHU7ykTV9U2zSaLdcGGBNAkZOLu52HJHqmzN3JjFwvHBa%2FHWJnsP1TZCCQZ5FMfPsUkrr9EI%2FLQwK3UAB%2FbGrBamtTEvl2I68MkuttRa5qpJ%2B74d1DA%2BrV075b5HIZaSmCzHkT%2B5UnRf6wc2QG0D2Pog1rvvwbAsDZog8VVY6ERLIUF99gjG6pXR281DZY8wq1d0pYcD%2BVCK%2FgTqZsRPugMjSFwwntJnh0xRMn4vDduqIgb6b6zcaqRiasoRNgRfONgfe3V2j4D9nbMvfHUEQMc3tdAc4XfqmQ2C0IpgLoeqK0Auz8c2l6DIbCVVN2rpZimAzrCthUnYLcGMuYYjCS%2Bo2vBjqwAcdKk%2BExWCTaz2mxM6MHtwClyTIzHKK0HFOsNSIeZcVAVmq%2FoXnDzRY%2BR76s4eshpmlIhB9oDloG8zHTU4Ybt1mma8ocqK8ehyyZKh7Ah2KlzK2xg%2B9hj%2BSNfNfmBjM%2B4Q5y6YRA00%2FcWJ2iDR%2F01Gu9PQv9boS2Y0oFG1zXPE4Gy4pnF7Td%2Fu25UKpyrOd6bm%2BTlRA3NjLF5xf1IdMmbquoUgTSLZ5exV2l1mETLjSt&Expires=1709538295 [following]\n",
      "--2024-03-02 22:44:56--  https://encode-public.s3.amazonaws.com/2023/01/15/5787db88-2846-4a48-8eb5-0bafa02771c9/ENCFF285FHP.bam?response-content-disposition=attachment%3B%20filename%3DENCFF285FHP.bam&AWSAccessKeyId=ASIATGZNGCNXY37BXRX4&Signature=rupRwOcUNWGjJM%2FO3TGddjOg%2B2w%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEFQaCXVzLXdlc3QtMiJIMEYCIQC6PdtSmfTx6QZwuEiXIb7ejZ%2F4TBmd8uEYX1Eb1h3sfAIhAOhudH49qg5H6faOR9aV4XeFxBK0ZpMF1Twg0sVw%2BPniKrMFCE0QABoMMjIwNzQ4NzE0ODYzIgzzwbFwvJgJ7AmXaxkqkAWVguA0uwr0o%2FOxIUBbsRpbw9xx7WSRaJ5o0x5c%2BfWUQgYzAS8eEZRZRQQIu7X7yADaXBMotYhTUCQNhUjasW4oT5RApJQ9XR857jMaSZxKskhhfKQP66giGFID5pxs4acJ2Z6A6WuT2oedcANoEG32EPZXLd2UBe66%2Bo9ahtxXkE5J%2B1Fu4IppNWRbnu4knxv7VY0AfnmVh8olWr6jC0Sr6axmm00SXagsbMNVv8z2DXg3x8L6Ha0lJMj7pLjqxxOXW%2ByEzboO9wPFb8jNMSs%2FGcnOAZu7t6eD9CnYZOS3GXGe%2Fn%2BItOqpmiwS%2F6KMx8TmzimZe8P9KjL2poD5SqGhGtL%2F8PcZLoHK3Ea4fiSI5NqJvKtVwNII%2FTQjixU5g7WEwLW8tq%2FODimrNGx3wT6fHdSZ9KBG6uACA1m0iWBAb7W40DkgfP2CKAcl7vYffkQ%2Fp1WxxZKNxeEIOH9nWBsUowjTF0J7g2Ei3mdp2rDvXMu1bgk9bdAS6LO76RB0uUH3xjaZo4RJhomRiaj%2FRTraN3WHU7ykTV9U2zSaLdcGGBNAkZOLu52HJHqmzN3JjFwvHBa%2FHWJnsP1TZCCQZ5FMfPsUkrr9EI%2FLQwK3UAB%2FbGrBamtTEvl2I68MkuttRa5qpJ%2B74d1DA%2BrV075b5HIZaSmCzHkT%2B5UnRf6wc2QG0D2Pog1rvvwbAsDZog8VVY6ERLIUF99gjG6pXR281DZY8wq1d0pYcD%2BVCK%2FgTqZsRPugMjSFwwntJnh0xRMn4vDduqIgb6b6zcaqRiasoRNgRfONgfe3V2j4D9nbMvfHUEQMc3tdAc4XfqmQ2C0IpgLoeqK0Auz8c2l6DIbCVVN2rpZimAzrCthUnYLcGMuYYjCS%2Bo2vBjqwAcdKk%2BExWCTaz2mxM6MHtwClyTIzHKK0HFOsNSIeZcVAVmq%2FoXnDzRY%2BR76s4eshpmlIhB9oDloG8zHTU4Ybt1mma8ocqK8ehyyZKh7Ah2KlzK2xg%2B9hj%2BSNfNfmBjM%2B4Q5y6YRA00%2FcWJ2iDR%2F01Gu9PQv9boS2Y0oFG1zXPE4Gy4pnF7Td%2Fu25UKpyrOd6bm%2BTlRA3NjLF5xf1IdMmbquoUgTSLZ5exV2l1mETLjSt&Expires=1709538295\n",
      "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.179.17, 52.92.235.145, 52.92.160.41, ...\n",
      "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.179.17|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 540683565 (516M) [binary/octet-stream]\n",
      "Saving to: ‘ENCFF285FHP.bam’\n",
      "\n",
      "ENCFF285FHP.bam     100%[===================>] 515.64M  4.39MB/s    in 2m 36s  \n",
      "\n",
      "2024-03-02 22:47:33 (3.30 MB/s) - ‘ENCFF285FHP.bam’ saved [540683565/540683565]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget https://www.encodeproject.org/files/ENCFF285FHP/@@download/ENCFF285FHP.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "!bedtools bamtobed -i ENCFF285FHP.bam > ENCFF285FHP.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2024-03-02 22:48:08--  https://www.encodeproject.org/files/ENCFF726NIH/@@download/ENCFF726NIH.bigWig\n",
      "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
      "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
      "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
      "Location: https://encode-public.s3.amazonaws.com/2023/01/15/9da46577-acd5-4b8c-a56f-73287c9e7202/ENCFF726NIH.bigWig?response-content-disposition=attachment%3B%20filename%3DENCFF726NIH.bigWig&AWSAccessKeyId=ASIATGZNGCNXY37BXRX4&Signature=ByCg%2BT2G6rcBM5EtpGmJ6e46DlQ%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEFQaCXVzLXdlc3QtMiJIMEYCIQC6PdtSmfTx6QZwuEiXIb7ejZ%2F4TBmd8uEYX1Eb1h3sfAIhAOhudH49qg5H6faOR9aV4XeFxBK0ZpMF1Twg0sVw%2BPniKrMFCE0QABoMMjIwNzQ4NzE0ODYzIgzzwbFwvJgJ7AmXaxkqkAWVguA0uwr0o%2FOxIUBbsRpbw9xx7WSRaJ5o0x5c%2BfWUQgYzAS8eEZRZRQQIu7X7yADaXBMotYhTUCQNhUjasW4oT5RApJQ9XR857jMaSZxKskhhfKQP66giGFID5pxs4acJ2Z6A6WuT2oedcANoEG32EPZXLd2UBe66%2Bo9ahtxXkE5J%2B1Fu4IppNWRbnu4knxv7VY0AfnmVh8olWr6jC0Sr6axmm00SXagsbMNVv8z2DXg3x8L6Ha0lJMj7pLjqxxOXW%2ByEzboO9wPFb8jNMSs%2FGcnOAZu7t6eD9CnYZOS3GXGe%2Fn%2BItOqpmiwS%2F6KMx8TmzimZe8P9KjL2poD5SqGhGtL%2F8PcZLoHK3Ea4fiSI5NqJvKtVwNII%2FTQjixU5g7WEwLW8tq%2FODimrNGx3wT6fHdSZ9KBG6uACA1m0iWBAb7W40DkgfP2CKAcl7vYffkQ%2Fp1WxxZKNxeEIOH9nWBsUowjTF0J7g2Ei3mdp2rDvXMu1bgk9bdAS6LO76RB0uUH3xjaZo4RJhomRiaj%2FRTraN3WHU7ykTV9U2zSaLdcGGBNAkZOLu52HJHqmzN3JjFwvHBa%2FHWJnsP1TZCCQZ5FMfPsUkrr9EI%2FLQwK3UAB%2FbGrBamtTEvl2I68MkuttRa5qpJ%2B74d1DA%2BrV075b5HIZaSmCzHkT%2B5UnRf6wc2QG0D2Pog1rvvwbAsDZog8VVY6ERLIUF99gjG6pXR281DZY8wq1d0pYcD%2BVCK%2FgTqZsRPugMjSFwwntJnh0xRMn4vDduqIgb6b6zcaqRiasoRNgRfONgfe3V2j4D9nbMvfHUEQMc3tdAc4XfqmQ2C0IpgLoeqK0Auz8c2l6DIbCVVN2rpZimAzrCthUnYLcGMuYYjCS%2Bo2vBjqwAcdKk%2BExWCTaz2mxM6MHtwClyTIzHKK0HFOsNSIeZcVAVmq%2FoXnDzRY%2BR76s4eshpmlIhB9oDloG8zHTU4Ybt1mma8ocqK8ehyyZKh7Ah2KlzK2xg%2B9hj%2BSNfNfmBjM%2B4Q5y6YRA00%2FcWJ2iDR%2F01Gu9PQv9boS2Y0oFG1zXPE4Gy4pnF7Td%2Fu25UKpyrOd6bm%2BTlRA3NjLF5xf1IdMmbquoUgTSLZ5exV2l1mETLjSt&Expires=1709538489 [following]\n",
      "--2024-03-02 22:48:09--  https://encode-public.s3.amazonaws.com/2023/01/15/9da46577-acd5-4b8c-a56f-73287c9e7202/ENCFF726NIH.bigWig?response-content-disposition=attachment%3B%20filename%3DENCFF726NIH.bigWig&AWSAccessKeyId=ASIATGZNGCNXY37BXRX4&Signature=ByCg%2BT2G6rcBM5EtpGmJ6e46DlQ%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEFQaCXVzLXdlc3QtMiJIMEYCIQC6PdtSmfTx6QZwuEiXIb7ejZ%2F4TBmd8uEYX1Eb1h3sfAIhAOhudH49qg5H6faOR9aV4XeFxBK0ZpMF1Twg0sVw%2BPniKrMFCE0QABoMMjIwNzQ4NzE0ODYzIgzzwbFwvJgJ7AmXaxkqkAWVguA0uwr0o%2FOxIUBbsRpbw9xx7WSRaJ5o0x5c%2BfWUQgYzAS8eEZRZRQQIu7X7yADaXBMotYhTUCQNhUjasW4oT5RApJQ9XR857jMaSZxKskhhfKQP66giGFID5pxs4acJ2Z6A6WuT2oedcANoEG32EPZXLd2UBe66%2Bo9ahtxXkE5J%2B1Fu4IppNWRbnu4knxv7VY0AfnmVh8olWr6jC0Sr6axmm00SXagsbMNVv8z2DXg3x8L6Ha0lJMj7pLjqxxOXW%2ByEzboO9wPFb8jNMSs%2FGcnOAZu7t6eD9CnYZOS3GXGe%2Fn%2BItOqpmiwS%2F6KMx8TmzimZe8P9KjL2poD5SqGhGtL%2F8PcZLoHK3Ea4fiSI5NqJvKtVwNII%2FTQjixU5g7WEwLW8tq%2FODimrNGx3wT6fHdSZ9KBG6uACA1m0iWBAb7W40DkgfP2CKAcl7vYffkQ%2Fp1WxxZKNxeEIOH9nWBsUowjTF0J7g2Ei3mdp2rDvXMu1bgk9bdAS6LO76RB0uUH3xjaZo4RJhomRiaj%2FRTraN3WHU7ykTV9U2zSaLdcGGBNAkZOLu52HJHqmzN3JjFwvHBa%2FHWJnsP1TZCCQZ5FMfPsUkrr9EI%2FLQwK3UAB%2FbGrBamtTEvl2I68MkuttRa5qpJ%2B74d1DA%2BrV075b5HIZaSmCzHkT%2B5UnRf6wc2QG0D2Pog1rvvwbAsDZog8VVY6ERLIUF99gjG6pXR281DZY8wq1d0pYcD%2BVCK%2FgTqZsRPugMjSFwwntJnh0xRMn4vDduqIgb6b6zcaqRiasoRNgRfONgfe3V2j4D9nbMvfHUEQMc3tdAc4XfqmQ2C0IpgLoeqK0Auz8c2l6DIbCVVN2rpZimAzrCthUnYLcGMuYYjCS%2Bo2vBjqwAcdKk%2BExWCTaz2mxM6MHtwClyTIzHKK0HFOsNSIeZcVAVmq%2FoXnDzRY%2BR76s4eshpmlIhB9oDloG8zHTU4Ybt1mma8ocqK8ehyyZKh7Ah2KlzK2xg%2B9hj%2BSNfNfmBjM%2B4Q5y6YRA00%2FcWJ2iDR%2F01Gu9PQv9boS2Y0oFG1zXPE4Gy4pnF7Td%2Fu25UKpyrOd6bm%2BTlRA3NjLF5xf1IdMmbquoUgTSLZ5exV2l1mETLjSt&Expires=1709538489\n",
      "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.179.17, 52.92.235.145, 52.92.160.41, ...\n",
      "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.179.17|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 1796877496 (1.7G) [binary/octet-stream]\n",
      "Saving to: ‘ENCFF726NIH.bigWig’\n",
      "\n",
      "ENCFF726NIH.bigWig  100%[===================>]   1.67G  3.03MB/s    in 8m 25s  \n",
      "\n",
      "2024-03-02 22:56:34 (3.40 MB/s) - ‘ENCFF726NIH.bigWig’ saved [1796877496/1796877496]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget https://www.encodeproject.org/files/ENCFF726NIH/@@download/ENCFF726NIH.bigWig"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Killed\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "35072"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# У меня не хватает памяти для постройки матрицы\n",
    "\n",
    "import os\n",
    "get_ipython().system = os.system\n",
    "!computeMatrix scale-regions \\\n",
    " -S ENCFF726NIH.bigWig \\\n",
    " -R ENCFF285FHP.bed \\\n",
    " -a 30 -b 30 \\\n",
    " -out matrix.tab.gz "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "usage: plotHeatmap [--matrixFile MATRIXFILE] --outFileName OUTFILENAME\n",
      "                   [--outFileSortedRegions FILE] [--outFileNameMatrix FILE]\n",
      "                   [--interpolationMethod STR] [--dpi DPI] [--kmeans KMEANS]\n",
      "                   [--hclust HCLUST] [--silhouette] [--help] [--version]\n",
      "                   [--plotType {lines,fill,se,std}]\n",
      "                   [--sortRegions {descend,ascend,no,keep}]\n",
      "                   [--sortUsing {mean,median,max,min,sum,region_length}]\n",
      "                   [--sortUsingSamples SORTUSINGSAMPLES [SORTUSINGSAMPLES ...]]\n",
      "                   [--linesAtTickMarks]\n",
      "                   [--clusterUsingSamples CLUSTERUSINGSAMPLES [CLUSTERUSINGSAMPLES ...]]\n",
      "                   [--averageTypeSummaryPlot {mean,median,min,max,std,sum}]\n",
      "                   [--missingDataColor MISSINGDATACOLOR]\n",
      "                   [--colorMap COLORMAP [COLORMAP ...]] [--alpha ALPHA]\n",
      "                   [--colorList COLORLIST [COLORLIST ...]]\n",
      "                   [--colorNumber COLORNUMBER] [--zMin ZMIN [ZMIN ...]]\n",
      "                   [--zMax ZMAX [ZMAX ...]] [--heatmapHeight HEATMAPHEIGHT]\n",
      "                   [--heatmapWidth HEATMAPWIDTH]\n",
      "                   [--whatToShow {plot, heatmap and colorbar,plot and heatmap,heatmap only,heatmap and colorbar}]\n",
      "                   [--boxAroundHeatmaps BOXAROUNDHEATMAPS]\n",
      "                   [--xAxisLabel XAXISLABEL] [--startLabel STARTLABEL]\n",
      "                   [--endLabel ENDLABEL] [--refPointLabel REFPOINTLABEL]\n",
      "                   [--labelRotation LABEL_ROTATION]\n",
      "                   [--regionsLabel REGIONSLABEL [REGIONSLABEL ...]]\n",
      "                   [--samplesLabel SAMPLESLABEL [SAMPLESLABEL ...]]\n",
      "                   [--plotTitle PLOTTITLE] [--yAxisLabel YAXISLABEL]\n",
      "                   [--yMin YMIN [YMIN ...]] [--yMax YMAX [YMAX ...]]\n",
      "                   [--legendLocation {best,upper-right,upper-left,upper-center,lower-left,lower-right,lower-center,center,center-left,center-right,none}]\n",
      "                   [--perGroup] [--plotFileFormat] [--verbose]\n",
      "plotHeatmap: error: argument --matrixFile/-m: can't open 'matrix.tab.gz': [Errno 2] No such file or directory: 'matrix.tab.gz'\n"
     ]
    }
   ],
   "source": [
    "!plotHeatmap \\\n",
    " -m matrix.tab.gz \\\n",
    " -out result.png \\\n",
    " --colorMap YlGnBu \\\n",
    " --regionsLabel 'genes' \\\n",
    " --heatmapHeight 15 \\\n",
    " --plotTitle 'ngs plot' &"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "bioinf",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
